system/ee/ExpressionEngine/Addons/search/mod.search.php

```diff
@@ -1013,10 +1013,18 @@ public function total_results()
         $query = ee()->db->query("SELECT total_results FROM exp_search WHERE search_id = '" . ee()->db->escape_str($search_id) . "'");
 
         if ($query->num_rows() == 1) {
-            return $query->row('total_results') ;
-        } else {
-            return 0;
+            if (empty(ee()->TMPL->tagdata)) {
+                return $query->row('total_results');
+            }
+
+            return $query->row('total_results') == 0
+            ? ee()->TMPL->no_results()
+                : ee()->TMPL->parse_variables_row(ee()->TMPL->tagdata, (array) $query->row(), true);
```

Comment:
No need to do typecasting for `(array) $query->row()`, you can just use `$query->row_array()`. Also since 3rd parameter is default, it can just be skipped (which is what we usually do)